src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -730,6 +733,36 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
     throw xrt_core::error("uuid does not match BLP");
   }
 
+  // -- process "user" option ---------------------------------------
+  if(!xclbin.empty()) {
+    XBU::verbose(boost::str(boost::format("  xclbin: %s") % xclbin));
+    //only 1 card and name
+    if(deviceCollection.size() > 1)
+      throw xrt_core::error("Please specify a single device");
+    auto dev = deviceCollection.front();
+
+    std::ifstream stream(xclbin, std::ios::binary);
+    if (!stream)
+      throw xrt_core::error(boost::str(boost::format("Could not open %s for reding") % xclbin));
+
+    stream.seekg(0,stream.end);
+    ssize_t size = stream.tellg();
+    stream.seekg(0,stream.beg);
+
+    std::vector<char> raw(size);
+    stream.read(raw.data(), size);
+    std::string v(raw.data(),raw.data()+7);
```

Comment:
Magic number '7'....I know it is lucky but why is it here :^).

Reply:
Also, what happens if the file to be read in is only 2 bytes in size.  In this case, you will be going beyond the buffer and the system "could" crash.  And yes, coverity will catch this issue.